### PR TITLE
fix(profile): Avoid re-enable for approval nfts

### PR DESCRIPTION
**Pull request by chefilip**
Avoid re-enabling NFT when re-selecting profile pic.

### CODE REVIEW

File: src/state/nftMarket/hooks.ts [hunk 1]
```diff
@@ -105,6 +109,35 @@ export const useUserNfts = (): UserNftsState => {
...
+    [nftsInWallet],
+  )
+
+  const { data } = useSWRMulticall(erc721Abi, nftApprovalCalls)
+
+  const approvedTokenIds = Array.isArray(data)
+    ? data
+        .flat()
+        .reduce(
+          (acc, address, index) => ({ ...acc, [nftsInWallet[index].tokenId]: getPancakeProfileAddress() === address }),
```
Comment (memoyil) [inline src/state/nftMarket/hooks.ts:133]: This should check against user address no?

Comment (chefilip) [reply src/state/nftMarket/hooks.ts:133]: `nftsInWallet` is already owned by user address. This check is for NFTs enabled to be used in Profile contract.

Comment (memoyil) [reply src/state/nftMarket/hooks.ts:133]: This i agree, but contract returns the address that approves the token not the pancake profile address afaik. Correct me if i am wrong

Comment (chefilip) [reply src/state/nftMarket/hooks.ts:127]: Screen Shot 2022-01-20 at 20 32 47

The approve logic is `approvalList[tokenId] = profileAddress`. Then the `getApprove` function would return `profileAddress` by `tokenId` 

You could check the contract for more detail. https://bscscan.com/address/0xDf7952B35f24aCF7fC0487D01c8d5690a60DBa07#code

Comment (memoyil) [reply src/state/nftMarket/hooks.ts:127]: Sorry my bad :) I confused my address with profile address. :)

File: src/state/nftMarket/hooks.ts [hunk 2]
```diff
@@ -99,6 +103,35 @@ export const useGetNFTInitializationState = () => {
...
+        return {
+          address: collectionAddress,
+          name: 'getApproved',
+          params: [tokenId],
+        }
+      }),
+    [nftsInWallet],
+  )
+
+  const { data } = useSWRMulticall(erc721Abi, nftApprovalCalls)
```
Comment (memoyil) [inline src/state/nftMarket/hooks.ts:121]: What is the advantage of using swr here?

File: src/hooks/useApproveConfirmTransaction.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import React, { useEffect, useReducer, useRef } from 'react'
-import noop from 'lodash/noop'
+import React, { useEffect, useReducer, useRef, useCallback } from 'react'
+import { noop } from 'lodash'
```
Comment (chef-jojo) [inline src/hooks/useApproveConfirmTransaction.tsx:2]: Import path

File: src/hooks/useSWRContract.ts [hunk 4]
```diff
@@ -120,6 +121,10 @@ export function useSWRContract<
   )
 }
 
+export function useSWRMulticall(abi: any[], calls: Call[], options: MulticallOptions = { requireSuccess: true }) {
+  return useSWR(calls, () => multicallv2(abi, calls, options))
```
Comment (chef-jojo) [inline src/hooks/useSWRContract.ts:125]: Use this in caution with multicall. By default, it refetch when on focus or mount.

File: src/state/nftMarket/hooks.ts (hunk 2 above)
Comment (chef-jojo) [reply src/state/nftMarket/hooks.ts:121]: Will fetch by selectedNft affect user experience when switching selected? Just wondering if it is good enough to not use multicall here

File: src/hooks/useApproveConfirmTransaction.tsx (hunk 3 above)
Comment (chefilip) [reply src/hooks/useApproveConfirmTransaction.tsx:2]: Oops. It seems prettier auto updates this part. Will check

Comment (chefilip) [reply src/hooks/useApproveConfirmTransaction.tsx:2]: Oops. Noted

File: src/hooks/useSWRContract.ts (hunk 4 above)
Comment (chefilip) [reply src/hooks/useSWRContract.ts:125]: Noted

File: src/state/nftMarket/hooks.ts (hunk 2 above)
Comment (chefilip) [reply src/state/nftMarket/hooks.ts:121]: Without multicall, whenever user selects a pic, a loading status is shown, which is not a good UX. For this feature, it's reasonable to call it in one batch.

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**